src/client/options/mod.rs

```diff
@@ -675,43 +675,181 @@ impl Serialize for ClientOptions {
+    ///
+    /// The default value is true.
     pub retry_reads: Option<bool>,
+
+    /// Whether or not the client should retry a write operation if the operation fails.
+    ///
+    /// The default value is true.
     pub retry_writes: Option<bool>,
-    pub socket_timeout: Option<Duration>,
-    pub zlib_compression: Option<i32>,
+
+    /// Specifies whether the Client should directly connect to a single host rather than
+    /// autodiscover all servers in the cluster.
+    ///
+    /// The default value is false.
     pub direct_connection: Option<bool>,
+
+    /// The credential to use for authenticating connections made by this client.
     pub credential: Option<Credential>,
+
+    /// Default database for this client.
+    ///
+    /// By default, no default database is specified.
     pub default_database: Option<String>,
-    max_staleness: Option<Duration>,
+
+    /// Whether or not the client is connecting to a MongoDB cluster through a load balancer.
+    pub load_balanced: Option<bool>,
+
+    /// Amount of time spent attempting to send or receive on a socket before timing out; note that
+    /// this only applies to application operations, not SDAM.
+    pub socket_timeout: Option<Duration>,
+
+    /// Specifies the level of compression when using zlib to compress wire protocol messages; -1
+    /// signifies the default level, 0 signifies no compression, 1 signifies the fastest speed, and
+    /// 9 signifies the best compression.
+    pub zlib_compression: Option<i32>,
+
+    /// The maximum replication lag, in wall clock time, that a secondary can suffer and still be
+    /// eligible for server selection.
+    pub max_staleness: Option<Duration>,
+
+    /// The authentication mechanism method to use for connection to the server.
+    pub auth_mechanism: Option<AuthMechanism>,
+
+    /// The database that connections should authenticate against.
+    pub auth_source: Option<String>,
+
+    /// Additional options provided for authentication (e.g. to enable hostname canonicalization
+    /// for GSSAPI).
+    pub auth_mechanism_properties: Option<Document>,
+
+    /// Default read preference for the client (excluding tags).
+    pub read_preference: Option<ReadPreference>,
+
+    /// Default read preference tags for the client; only valid if the read preference mode is not
+    /// primary.
+    ///
+    /// The order of the tag sets in the read preference is the same as the order they are
+    /// specified in the URI.
+    pub read_preference_tags: Option<Vec<TagSet>>,
+
+    pub(crate) srv: bool,
+    wait_queue_timeout: Option<Duration>,
     tls_insecure: Option<bool>,
-    auth_mechanism: Option<AuthMechanism>,
-    auth_source: Option<String>,
-    auth_mechanism_properties: Option<Document>,
-    read_preference: Option<ReadPreference>,
-    read_preference_tags: Option<Vec<TagSet>>,
-    load_balanced: Option<bool>,
     original_uri: String,
```

Comment:
I think storing this here might actually be a bit dangerous, since users could mutate this connection string via its fields and create a disconnect. I think we'll instead want to implement `Display` on `ConnectionString` and use that implementation if we need to get the original / current URI.

Do you have any idea how much work that would constitute @isabelatkinson? If it's a lot, we could break it (and `Serialize`) implementations out into a separate ticket, seeing as this is a quick win.

Reply:
[This](https://github.com/mongodb/mongo-swift-driver/blob/08d68a5c863d10da7221880a6a98e758a99bb63d/Sources/MongoSwift/MongoConnectionString.swift#L931) is the equivalent implementation in Swift -- it's not a ton of work but it is pretty tedious. I think it would be fine to do in a separate ticket (I'm pretty sure I did that implementation as its own PR) as long as it gets done prior to us releasing this work.

Reply:
Makes sense; I filed [RUST-1308](https://jira.mongodb.org/browse/RUST-1308) for that.  `original_uri` is used in a lot of tests so I didn't want to remove it outright, but I added `#[cfg(test)]` to enforce that it's not part of internal or external API.